R/R/install.R

```diff
@@ -0,0 +1,89 @@
+#' Install Metaflow Python package
+#'
+#' This function wraps installation functions from [reticulate][reticulate::reticulate] to install the Python packages
+#' **metaflow** and it's Python dependencies.
+#'
+#' This package uses the [reticulate][reticulate::reticulate] package
+#' to make an interface with the [Metaflow](https://metaflow.org/)
+#' Python package.
+#'
+#' @param method `character`, indicates to use `"conda"` or `"virtualenv"`.
+#' @param version `character`, version of Metaflow to install. The default version
+#' is the latest available on PyPi.
+#' @param ... other arguments sent to [reticulate::conda_install()] or
+#'    [reticulate::virtualenv_install()]
+#'
+#' @seealso
+#' [reticulate: Using reticulate in an R Package](https://rstudio.github.io/reticulate/articles/package.html),
+#' [reticulate: Installing Python Packages](https://rstudio.github.io/reticulate/articles/python_packages.html)
+#' @examples
+#' \dontrun{
+#' # not run because it requires Python
+#' install_metaflow()
+#' }
+#' @export
+install_metaflow <- function(method = c("conda", "virtualenv"),
+                             version = NULL,
+                             ...) {
+  envname <- "metaflow-r"
+
+  # validate stage, method arguments
+  method <- match.arg(method)
+
+  # conda and pip use different syntax for indicating versions
+  if (identical(method, "conda")) {
+    version_sep <- "="
+  } else {
+    version_sep <- "=="
+  }
+
+  if (is.null(version)) {
+    metaflow_pkg_version <- "metaflow"
+  } else {
+    metaflow_pkg_version <- paste("metaflow", version, sep = version_sep)
+  }
+
+  packages <- c(metaflow_pkg_version, "numpy", "pandas")
+
+  # create environment if not present
+  if (method == "conda" && !envname %in% conda_list()$name) {
+    conda_create(envname)
+  } else if (method == "virtualenv" && !envname %in% virtualenv_list()) {
+    virtualenv_create(envname)
+  }
+
+  py_install(
```

Comment:
Does this function automatically infer whether to use `conda` or `pip` to install dependencies?

Reply:
Currently it uses the defaults for `conda_create`, which is `pip=FALSE`. Should probably pass `...`s here or (add pip as an argument) to make it more flexible.

Reply:
Great!